app/uk/gov/hmrc/bindingtariffclassification/service/CaseService.scala

```diff
@@ -19,19 +19,19 @@ package uk.gov.hmrc.bindingtariffclassification.service
 import java.util.UUID
 
 import javax.inject._
-import uk.gov.hmrc.bindingtariffclassification.model.{Case, CaseStatusChange, Event}
 import uk.gov.hmrc.bindingtariffclassification.model.CaseStatus.CaseStatus
 import uk.gov.hmrc.bindingtariffclassification.model.search.CaseParamsFilter
-import uk.gov.hmrc.bindingtariffclassification.repository.CaseRepository
+import uk.gov.hmrc.bindingtariffclassification.model.{Case, CaseStatusChange, Event}
+import uk.gov.hmrc.bindingtariffclassification.repository.{CaseRepository, SequenceRepository}
 
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 
 @Singleton
-class CaseService @Inject()(repository: CaseRepository, eventService: EventService) {
+class CaseService @Inject()(caseRepository: CaseRepository, sequenceRepository: SequenceRepository, eventService: EventService) {
```

Comment:
probably you have thought about that one, but maybe create a SequenceService class we could abstract the meaning of a Sequence (easy to test and prove) to all the services that will use it (but most likely will be just this one).

Reply:
You will need to make sure that the `getAndIncrement` operation is atomic

Reply:
I did consider it but I didn't know what it would do. Maybe it would make more sense to have a `CaseReferenceService` but I figured that logic can just reside in the `CaseService` until it needs to be pulled out.